src/main/java/com/gregtechceu/gtceu/common/item/tool/behavior/MetaMachineConfigCopyBehaviour.java

```diff
@@ -0,0 +1,204 @@
+            stack.getOrCreateTag().put(CONFIG_DATA, configData);
+        }
+        return InteractionResult.SUCCESS;
+    }
+
+    public static InteractionResult handlePaste(ItemStack stack, MetaMachine machine) {
+        if (!stack.hasTag() || !stack.getTag().contains(CONFIG_DATA)) return InteractionResult.PASS;
+        CompoundTag root = stack.getOrCreateTag();
+        CompoundTag configData = root.getCompound(CONFIG_DATA);
+        Direction originalFront = intToDirection(configData.getInt(ORIGINAL_FRONT));
+        if (configData.contains(ITEM_CONFIG) && machine instanceof IAutoOutputItem autoOutputItem) {
+            CompoundTag itemData = configData.getCompound(ITEM_CONFIG);
+            autoOutputItem.setOutputFacingItems(getRelativeDirection(originalFront, machine.getFrontFacing(),
+                    intToDirection(itemData.getInt(DIRECTION))));
+            autoOutputItem.setAutoOutputItems(itemData.getBoolean(AUTO));
+            autoOutputItem.setAllowInputFromOutputSideItems(itemData.getBoolean(INPUT_FROM_OUTPUT_SIDE));
+        }
+        if (configData.contains(FLUID_CONFIG) && machine instanceof IAutoOutputFluid autoOutputFluid) {
+            CompoundTag fluidData = configData.getCompound(FLUID_CONFIG);
+            autoOutputFluid.setOutputFacingFluids(getRelativeDirection(originalFront, machine.getFrontFacing(),
+                    intToDirection(fluidData.getInt(DIRECTION))));
+            autoOutputFluid.setAutoOutputFluids(fluidData.getBoolean(AUTO));
+            autoOutputFluid.setAllowInputFromOutputSideFluids(fluidData.getBoolean(INPUT_FROM_OUTPUT_SIDE));
+        }
+        if (configData.contains(MUFFLED) && machine instanceof IMufflableMachine mufflableMachine) {
+            mufflableMachine.setMuffled(configData.getBoolean(MUFFLED));
+        }
+        return InteractionResult.SUCCESS;
+    }
+
+    @Override
+    public void appendHoverText(ItemStack stack, @Nullable Level level, List<Component> tooltipComponents,
+                                TooltipFlag isAdvanced) {
+        tooltipComponents.add(Component.translatable("behaviour.meta.machine.config.copy.tooltip"));
+        tooltipComponents.add(Component.translatable("behaviour.meta.machine.config.paste.tooltip"));
+        if (!stack.hasTag()) return;
+        if (Screen.hasShiftDown()) {
+            tooltipComponents.add(Component.literal(""));
+            var tag = stack.getOrCreateTag();
+            if (tag.contains(CONFIG_DATA)) {
+                var data = tag.getCompound(CONFIG_DATA);
+                if (data.contains(ORIGINAL_FRONT)) {
+                    var origFront = intToDirection(data.getInt(ORIGINAL_FRONT));
+                    if (data.contains(ITEM_CONFIG)) {
+                        var itemData = data.getCompound(ITEM_CONFIG);
+                        tooltipComponents.add(Component.translatable("behaviour.setting.output.direction.tooltip",
+                                "§6Item§r",
```

Comment:
please change all these literal texts into lang for translation purposes

Reply:
done